src/StringTools/StringTools.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.StringTools
+{
+    public static class Strings
```

Comment:
Nit re naming, `Strings.Intern(xyz)` might be easily confused (or mistyped) for `String.Intern(xyz)` which is rather different and would cause a leak.

Reply:
Agreed. I have renamed the method to `WeakIntern`.